mesonbuild/wrap/wrap.py

```diff
@@ -164,17 +164,20 @@ def resolve_git_submodule(self, dirname):
         if not ret:
             return False
         # Submodule has not been added, add it
-        if out.startswith(b'-'):
+        if out.startswith(b'+'):
+            mlog.warning('submodule {} might be out of date'.format(dirname))
+            return True
+        elif out.startswith(b'U'):
+            raise RuntimeError('submodule {} has merge conflicts'.format(dirname))
+        elif out.startswith(b'-'):
             if subprocess.call(['git', '-C', self.subdir_root, 'submodule', 'update', '--init', dirname]) != 0:
                 return False
```

Comment:
It looks like if the `git submodule update` succeeds, this now drops down into the (now unconditional) `Unknown git submodule output` below

Reply:
Okay that's dumb, I'm going to fix this and add a test for it...

Filed https://github.com/mesonbuild/meson/issues/3088.